pyiron_atomistics/sphinx/base.py

```diff
@@ -1702,6 +1707,126 @@ def check_vasp_potentials():
             ]
         )
 
+    def run_addon(self, addon, args=None, from_tar=None,
+                  silent=False, log=True,
+                  module_version=Automatic,
+                  debug=False):
+        """ Run a SPHInX addon
+
+            addon          - name of addon (str)
+            args           - arguments (str or list)
+            from_tar       - if job is compressed, extract these files (list)
+            silent         - do not print output for successful runs?
+            log            - produce log file?
+            module_version - which sphinx version to load (str or None)
+            debug          - return subprocess.CompletedProcess ?
+
+        """
+        if self.is_compressed () and from_tar is None:
+            raise FileNotFoundError (
+                "Cannot run add-on on compressed job without 'from_tar' parameter.\n"
+               +"   Solution 1: Run .decompress () first.\n"
+               +"   Solution 2: specify from_tar list to run in temporary directory\n"
+               +"   Solution 3: run with from_tar=[] if no files from tar are needed\n"
+                )
+
+        sxversion = 'sphinx'
+        if isinstance(module_version,str):
+            if "sphinx" not in module_version:
+                sxversion += '/' + module_version
+            else:
+                sxversion = module_version
+        elif module_version is Automatic:
+            # TODO: find a way to determine from self.executable
+            out = subprocess.run ("module load sphinx",
+                                  cwd=self.working_directory,
+                                  shell=True, stdout=PIPE, stderr=PIPE)
```

Comment:
This is rather cluster specific, is it possible to move this to the pyiron resources?

Reply:
agreed in principle.
However, I refuse to use the job.Executable mechanism because there are dozens of possible add-ons, and the module versions are very dynamic. I am not going to create hundreds of wrapper scripts for that purpose. The entire run_addon idea is to leverage that complexity.

Rough idea:

Approach will be to have a get_sx_versions script that generates the info dynamically; this will be located in the sphinx/ subfolder of the ressource_path and hence know about the specific requirements of the cluster.
How do I put a "default" get_sx_versions for the standard pyiron installation? I.e., where does the anaconda share/pyiron/sphinx/ folder come from? 
The one for the cmmc will be created separately, and I know how to do that.

The info I need will be pairs of <VERSION> <PATH-SETTING COMMAND> - this is essentially dictionary-style data.
Is there a prefered format for config files for such data? I can do
<VERSION>=<COMMAND>
or
json 
{ version: "<VERSION>", pathcommand: "<COMMAND>" }
or anything else.

For the cluster, this will be simply
2.6.3=module load sphinx/2.6.3
3.0.8-openmp=module load sphinx/3.0.8-openmp
etc.

For addons in the standard path, this could be
default=

For addons with a specific path, this could be
devel=export PATH=$HOME/devel/sphinx-debug/bin/:${PATH}

so there's a whole lot of flexibility of how to enable different code versions.

Reply:
@freyso An alternative idea would be to provide these utilities as part of the `sphinxdft` conda package, then we do not need a separate module load and other people who just test S/PHI/nX also have access to them. Are they included in our current package? Is it difficult to include them? Are they performance critical?

Reply:
I critically need the versioning for the addons.
Some add-ons may be performance-sensitive. Some add-ons change over time (added or removed features). And the wavefunctions can have hdf5 format or netcdf format, and this is compiled into the code (which is one of the main motivations for this entire branch).
The current solution is 100% transparent, all versioning is provided by the local installation, no hardcoded/defaulted versions from pyiron.

Reply:
but yes, all SPHInX addons should be part of the sphinxdft conda package. So in principle, they are available to testers, too (provided that the search paths are properly set), and many of them are working across different SPHInX versions.

Reply:
> I critically need the versioning for the addons.

I agree - I am just afraid that it becomes a sphinx specific solution.

Reply:
Well, for the time being, it is a SPHInX-specific problem...
It's probably related to the fact that I am currently the only one who is trying to do active code development while using pyiron. So my need for versioning is much higher than for most others, and my requirements for having full access to even the most bizarre features of the code as well.
Please give me some freedom in trying out possible solutions in this corner.